tests/containers/kubectl.pm

```diff
@@ -0,0 +1,69 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Test the kubectl utility
+# - List pods
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use utils;
+use version_utils;
+use publiccloud::utils;
+use registration qw(cleanup_registration register_product add_suseconnect_product get_addon_fullname remove_suseconnect_product);
+use containers::k8s;
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    # kubectl is in the container module (on SLES)
+    add_suseconnect_product(get_addon_fullname('contm')) if (is_sle);
+    zypper_call("-n in kubernetes-client");
+
+    # Record kubectl version and check if the tool itself is healthy
+    record_info("kubectl", script_output("kubectl version --client --output=json"));
+
+    # Configure CSP/k3s. Only one CSP at a time is possible due to the conflicting configuration in .kube/config
+    assert_script_run('mkdir -p ~/.kube');
+    my $provider = get_var("KUBECTL_CLUSTER", "k3s");
+    if ($provider eq "AKS") {
```

Comment:
Not sure we could use AKS after seeing the checks (e.g. check if there are not ingresses). Maybe for this test should be removed, or the checks changes. In public kubernetes we could have concurrent tests that could create these resources and the test will fail. An other option is to create a namespace and check these things inside the namespace. Deleting the namespace at the end

Reply:
I'm a bit lost here - where is the conflict?

We for sure need to adjust some of the checks later, but AFAICS the whole `kubectl` test run is read-only and does not create nor delete nor modify any resources?

Reply:
Wait, I think I know what you mean. This was meant as a template for future extensions. I commented it out and added a note to mark it as such. This is not (yet) intended to be used within the test code.